.github/workflows/deploy.yml

```diff
@@ -0,0 +1,158 @@
+---
+name: deploy
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-html:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: guides
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Ruby 2.6
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.6.x
+
+      - name: Setup Python 2.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.x'
+
+      - name: Install AsciiDoctor
+        run: gem install asciidoctor
+
+      - name: Install linkchecker
+        run: pip install linkchecker
+
+      - name: Clean the environment
+        run: make clean
+
+      - name: Build HTML
+        run: |
+          make html BUILD=foreman
+          make html BUILD=foreman-deb
+          make html BUILD=satellite
+
+      - name: Upload HTML
+        uses: actions/upload-artifact@v2
+        with:
+          name: foreman-docs-html
+          path: guides/build/
+
+      - name: Check HTML links
+        run: |
+          make linkchecker BUILD=foreman
+          #make linkchecker BUILD=foreman-deb
+          #make linkchecker BUILD=satellite
```

Comment:
This will be uncommented after I confirm this works fine. It's very slow, same below.